aiohttp/client_reqrep.py

```diff
@@ -103,7 +106,8 @@ def __init__(self, method, url, *,
         self.update_proxy(proxy, proxy_auth, proxy_from_env)
 
         self.update_body_from_data(data)
-        self.update_transfer_encoding()
+        if data or self.method not in self.GET_METHODS:
```

Comment:
Hm, ok. Asking because you can send POST without content-length set and that's fine as well. curl does so.